### PR TITLE
Ut for order methods

### PR DESCRIPTION
**Pull request by garimaprachi**
Added unit test for getRazorpayPaymentParams, getCustomOrdercreationMessage and getDefaultCheckoutArguments

### CODE REVIEW

File: tests/phpunit/tests/test-order-methods.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        $this->instance->shouldReceive('getRazorpayApiInstance')->andReturnUsing(
+            function () {
+                return new MockApi('key_id_2', 'key_secret2');
+            });
+
+        $this->instance->shouldReceive('autoEnableWebhook');
+        
+        $razorpayOrderId=$this->instance->shouldReceive('createOrGetRazorpayOrderId')->with($wcOrderId)->andReturn('order_test');
+        
+        $this->assertEquals(['order_id'  =>  'order_test'] , $this->instance->getRazorpayPaymentParams($wcOrderId));
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:42]: @garimaprachi set one white space before and after =>, and no space before comma

File: tests/phpunit/tests/test-order-methods.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+        $message = 'RAZORPAY ERROR: Razorpay API could not be reached';
+       
+        try {
+            
+            $this->instance->getRazorpayPaymentParams($wcOrderId);
+            
+            $this->fail("Expected Exception has not been raised.");
+        
+        } catch (Exception $ex) {
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:74]: @garimaprachi fix indentation and flower bracket

File: tests/phpunit/tests/test-order-methods.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    {
+        $order = wc_create_order();
+        
+        $orderId = $order->get_order_number();
+        
+        $wcOrderId = $order->get_id();
+        
+        $desc = "Order $orderId";
+        
+        $notes = array('woocommerce_order_id'=> $orderId, 'woocommerce_order_number'=> $wcOrderId );
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:105]: @garimaprachi leave a white space before =>

File: tests/phpunit/tests/test-order-methods.php [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        $wcOrderId = $order->get_id();
+
+        $this->instance->shouldReceive('getRazorpayApiInstance')->andReturnUsing(
+            function () {
+                return new MockApi('key_id_2', 'key_secret2');
+            });
+
+        $this->instance->shouldReceive('autoEnableWebhook');
+        
+        $razorpayOrderId=$this->instance->shouldReceive('createOrGetRazorpayOrderId')->with($wcOrderId)->andReturn('order_test');
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:42]: @garimaprachi add space before and after =

File: tests/phpunit/tests/test-order-methods.php [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+                return new MockApi('key_id_2', 'key_secret2');
+            });
+        
+        $this->instance->shouldReceive('autoEnableWebhook');
+
+        $razorpayOrderId=$this->instance->shouldReceive('createOrGetRazorpayOrderId')->with($wcOrderId)->andReturn(null);
+
+        $message = 'RAZORPAY ERROR: Razorpay API could not be reached';
+       
+        try {
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:70]: @garimaprachi flower bracket should start from new line

File: tests/phpunit/tests/test-order-methods.php [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+            
+            $this->instance->getRazorpayPaymentParams($wcOrderId);
+            
+            $this->fail("Expected Exception has not been raised.");
+        
+        } catch (Exception $ex) {
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:76]: @garimaprachi flower bracket should start with new line and fix indentation

File: tests/phpunit/tests/test-order-methods.php [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    {
+        $order = wc_create_order();
+        
+        $orderId = $order->get_order_number();
+        
+        $wcOrderId = $order->get_id();
+        
+        $desc = "Order $orderId";
+        
+        $notes = array('woocommerce_order_id' => $orderId, 'woocommerce_order_number' => $wcOrderId );
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:106]: @garimaprachi remove space before )

File: tests/phpunit/tests/test-order-methods.php [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+        $this->instance->shouldReceive('autoEnableWebhook');
+
+        $razorpayOrderId = $this->instance->shouldReceive('createOrGetRazorpayOrderId')->with($wcOrderId)->andReturn('order_test');
+
+        $this->assertEquals(['order_id' => 'order_test'], $this->instance->getRazorpayPaymentParams($wcOrderId));
+
+        $getWebhookFlag = 2500;
+
+        $this->assertEquals(['order_id' => 'order_test'], $this->instance->getRazorpayPaymentParams($wcOrderId));
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:48]: @garimaprachi why we have same assertion twice

Comment (garimaprachi) [reply tests/phpunit/tests/test-order-methods.php:48]: Added comment

File: tests/phpunit/tests/test-order-methods.php [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+        $order->set_address($address, 'billing');
+
+        $args = array(
+            'name'    => $order->get_billing_first_name() . ' ' . $order->get_billing_last_name(),
+            'email'   => $order->get_billing_email(),
+            'contact' => $order->get_billing_phone(),
+        );
+
+        $this->instance->expects($this->once())->method('getRedirectUrl')->with($orderId)->andReturn($this->returnValue($callbackUrl));
```
Comment (abdulwahidsharief) [inline tests/phpunit/tests/test-order-methods.php:133]: @garimaprachi undefined variable $callbackUrl

Comment (garimaprachi) [reply tests/phpunit/tests/test-order-methods.php]: Removed mocking as it is not required

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

**Pull request merged**